projects/timer_challenge/src/main.c

```diff
@@ -0,0 +1,47 @@
+#include <stdint.h>  // for integer types
+
+#include "interrupt.h"   // interrupts are required for soft timers
+#include "log.h"         // for printing
+#include "soft_timer.h"  // for soft timers
+#include "wait.h"        // for wait function
+
+#define COUNTER_A_INCREMENT_INTERVAL_MS 500
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void timer_callback(SoftTimerId timer_id, void *context) {
```

Comment:
Since this is a private function, it should be `static` and its name should be prefixed with `prv_`. `static` means that the function can't be seen by any other file; without `static`, if another file tried to define a `timer_callback` function, it might cause issues. The `prv_` is just a convention ([and you can see the full list here!](https://uwmidsun.atlassian.net/wiki/spaces/ELEC/pages/9306218/Coding+Standards)).